From 86692366b7edbd6dd1ce329a172fb78d402ac328 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Wed, 9 Mar 2016 11:01:43 +0000
Subject: [PATCH] Resolves: rhbz#1315385 use preferred size if widget supports
 it

when deciding if a popup needs to be placed up or down to stay
visible on screen

Change-Id: I718e0ee4a79152e919ac95841e15d4b53764ac78
---
 vcl/source/window/floatwin.cxx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vcl/source/window/floatwin.cxx b/vcl/source/window/floatwin.cxx
index 00c3b34..6553f94 100644
--- a/vcl/source/window/floatwin.cxx
+++ b/vcl/source/window/floatwin.cxx
@@ -244,7 +244,7 @@ Point FloatingWindow::ImplCalcPos( vcl::Window* pWindow,
 {
     // get window position
     Point       aPos;
-    Size        aSize = pWindow->GetSizePixel();
+    Size        aSize = ::isLayoutEnabled(pWindow) ? pWindow->get_preferred_size() : pWindow->GetSizePixel();
     Rectangle   aScreenRect = pWindow->ImplGetFrameWindow()->GetDesktopRectPixel();
     FloatingWindow *pFloatingWindow = dynamic_cast<FloatingWindow*>( pWindow );
 
-- 
2.7.1

